transport/stream.go

```diff
@@ -1,10 +1,40 @@
 package transport
 
 import (
+	"fmt"
+	"math"
+	"math/bits"
 	"scylla-go-driver/frame"
 )
 
 type StreamIDAllocator interface {
 	Alloc() (frame.StreamID, error)
 	Free(id frame.StreamID)
 }
+
+const maxStreamID = math.MaxInt16
+
+const bucketSize = 64
+const buckets = (maxStreamID + 1) / bucketSize
+
+// DefaultStreamIDAllocator is a StreamIDAllocator that always allocates the smallest possible stream on Alloc().
+type DefaultStreamIDAllocator struct {
+	usedBitmap [buckets]uint64
+}
+
+func (s *DefaultStreamIDAllocator) Alloc() (frame.StreamID, error) {
+	for blockID, block := range &s.usedBitmap {
```

Comment:
Why address &?

Reply:
This is because you are using an array! If you were using a slice it would be fine.

Reply:
The linter is correct.